design-system-docs/src/_component_docs/date-input.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Date input
+---
+
+A date input field allows the user to enter a date they remember.
+
+## Examples
+
+### Default
+
+<%= render(Shared::ComponentExample.new(:date_input, :default)) %>
+
+### Optional
+
+<%= render(Shared::ComponentExample.new(:date_input, :optional)) %>
+
+### With hint
+
+<%= render(Shared::ComponentExample.new(:date_input, :with_hint)) %>
+
+### With error message
+
+<%= render(Shared::ComponentExample.new(:date_input, :with_error)) %>
+
+### With incomplete error
+
+<%= render(Shared::ComponentExample.new(:date_input, :with_incomplete_error)) %>
+
+### With validation error
+
+<%= render(Shared::ComponentExample.new(:date_input, :with_validation_error)) %>
+
+### With values
+
+<%= render(Shared::ComponentExample.new(:date_input, :with_values)) %>
+
+## Using with Rails
```

Comment:
My personal opinion is that the "Using with Rails" section is probably a bit redundant because it often matches information that users can get from the examples "ERB code". This is just an observation and we should have a separate discussion and potentially PR about it.